src/levelset/levelSetSegmentationObject.cpp

```diff
@@ -1817,6 +1817,92 @@ LevelSetIntersectionStatus LevelSetSegmentationBaseObject::_isCellIntersected(lo
     return LevelSetObject::_isCellIntersected(id, distance, mode);
 }
 
+/*!
+ * Check if the specified interface intersects the zero-levelset iso-surface.
+ *
+ * The iso-surface is considered planar. Calculate the intersection
+ * as a segment defined by two points and the part of the interface belonging to the subspace
+ * of negative level-set. If the interface is not intersected, an empty polygon is returned. 
+ * The interface belonging to the possitive level-set subspace is returned if the
+ * "positivePart" argument is true.
+ *
+ * The intersection is computed by considering the zero-levelset as a curved surface.
+ *
+ * @param[in] id is the interface index
+ * @param[in] positivePart when true the part of the  interface occupying the positive levelset area is returned
+ * @param[in] tolerance is the tolerance used for distance comparisons
+ * @param[out] intersection If intersects, return the cutting segment. In case of a liner interface
+ * the intersection degenerates to a point which coincides with the begining and the ending
+ * of the segment
+ * @param[out] polygon is the polygon constructed by the interface intersection placed on the
+ * opposite subspace of the one pointed by the plane's normal
+ * @return the level set status
+ */
+LevelSetIntersectionStatus LevelSetSegmentationBaseObject::_isInterfaceIntersected(long id, bool positivePart,
+                                                                                   double tolerance,
```

Comment:
Get the tolerance using the levelset function "double LevelSetKernel::getDistanceTolerance() const".

Reply:
Done